pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -476,38 +486,57 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 				s.addCon(con.ConID, con)
 				defer s.removeCon(con.ConID, con)
 			}
-		case <-con.pushChannel:
+		case pushEv, _ := <-con.pushChannel:
 			// It is called when config changes.
 			// This is not optimized yet - we should detect what changed based on event and only
 			// push resources that need to be pushed.
-			if con.CDSWatch {
-				err := s.pushCds(con)
-				if err != nil {
-					return err
-				}
-			}
-			if len(con.Routes) > 0 {
-				err := s.pushRoute(con)
-				if err != nil {
-					return err
-				}
-			}
-			if len(con.Clusters) > 0 {
-				err := s.pushEds(con)
-				if err != nil {
-					return err
-				}
-			}
-			if con.LDSWatch {
-				err := s.pushLds(con)
-				if err != nil {
-					return err
-				}
+
+			err := s.pushAll(con, pushEv)
+			if err != nil {
+				return nil
```

Comment:
(reading through code and randomly noticed this) just to make sure, is it intentional that this returns `nil` instead of `err`?